test/org/zaproxy/zap/extension/ascanrules/ActiveScannerTest.java

```diff
@@ -0,0 +1,216 @@
+        }
+        Constant.messages.addMessageBundle("ascanrules", bundle );
+
+        PluginFactory pluginFactory = Mockito.mock(PluginFactory.class);
+        ScanPolicy scanPolicy = Mockito.mock(ScanPolicy.class);
+        Mockito.when(scanPolicy.getPluginFactory()).thenReturn(pluginFactory);
+        
+        ConnectionParam connectionParam = new ConnectionParam();
+        
+        ScannerParam scannerParam = new ScannerParam();
+        // Will need this once we go to the release after 2.5.0
+        // RuleConfigParam ruleConfigParam = new RuleConfigParam();
+        Scanner parentScanner =
+                new Scanner(scannerParam, connectionParam, scanPolicy);
+        //, ruleConfigParam);
+
+        int port = 9090;
+        nano = new HTTPDTestServer(port);
+        nano.start();
+        
+        alertsRaised = new ArrayList<>();
+        parent = new HostProcess(
+                "localhost:" + port,
+                parentScanner, 
+                scannerParam, 
+                connectionParam, 
+                scanPolicy) {
+                //ruleConfigParam) {
+            @Override
+            public void alertFound(Alert arg1) {
+                alertsRaised.add(arg1);
+            }
+        };
+        
+        rule = createScanner();
+    }
+    
+    @After
+    public void shutDown() throws Exception {
+        nano.stop();
+        File dir = new File("test/resources/home");
+        FileUtils.deleteDirectory(dir);
+    }
+
+    protected abstract AbstractPlugin createScanner();
+    
+    protected HttpMessage getHttpMessage(String url) throws HttpMalformedHeaderException {
+        return this.getHttpMessage("GET", url, "<html></html>");
+        
+    }
+    protected HttpMessage getHttpMessage(String method, String url, String body) throws HttpMalformedHeaderException {
+        HttpMessage msg = new HttpMessage();
+        StringBuilder reqHeaderSB = new StringBuilder();
+        reqHeaderSB.append(method);
+        reqHeaderSB.append(" http://localhost:");
+        reqHeaderSB.append(this.nano.getListeningPort()); 
+        reqHeaderSB.append(url);
+        reqHeaderSB.append(" HTTP/1.1\r\n");
+        reqHeaderSB.append("Host: www.any_domain_name.org\r\n");
+        reqHeaderSB.append("User-Agent: ZAP\r\n");
+        reqHeaderSB.append("Pragma: no-cache\r\n");
+        msg.setRequestHeader(reqHeaderSB.toString());
+
+        msg.setResponseBody(body);
+        
+        StringBuilder respHeaderSB = new StringBuilder();
+        respHeaderSB.append("HTTP/1.1 200 OK\r\n");
+        respHeaderSB.append("Server: Apache-Coyote/1.1\r\n");
+        respHeaderSB.append("Content-Type: text/html;charset=ISO-8859-1\r\n");
+        respHeaderSB.append("Content-Length: ");
+        respHeaderSB.append(msg.getResponseBody().length());
+        respHeaderSB.append("\r\n");
+        msg.setResponseHeader(respHeaderSB.toString());
+
+        return msg;
+    }
+    
+    public String getHtml (String name) {
```

Comment:
All of the `getHtml` functions seem to have an extra space between func name and brackets.